### PR TITLE
first working version for the mac app store

### PR DESCRIPTION
**Pull request by kof**

### CODE REVIEW

File: tasks/release/osx.js [hunk 1]
```diff
@@ -6,28 +6,78 @@ var jetpack = require('fs-jetpack');
...
 
 var init = function () {
     projectDir = jetpack;
     tmpDir = projectDir.dir('./tmp', { empty: true });
+    distDir = tmpDir.dir('./dist', { empty: true });
     releasesDir = projectDir.dir('./releases');
     manifest = projectDir.read('app/package.json', 'json');
-    electronVersion = projectDir.read('package.json', 'json').devDependencies['electron-prebuilt'];
+    electronVersion = projectDir.read('package.json', 'json').devDependencies.electron;
+    electronVersion = semver.clean(electronVersion.replace(/\^/g, ''))
```
Comment (giuseppeg) [inline tasks/release/osx.js]: the regex should be `/^[^0-9]+/`

File: tasks/release/osx.js [hunk 2]
```diff
@@ -6,28 +6,78 @@ var jetpack = require('fs-jetpack');
...
+
+    download({
+      version: electronVersion,
+      platform: 'mas',
+      arch: process.env.npm_config_arch,
+      strictSSL: process.env.npm_config_strict_ssl === 'true',
+      quiet: ['info', 'verbose', 'silly', 'http'].indexOf(process.env.npm_config_loglevel) === -1
+    }, extractFile);
+
+    // unzips and makes path.txt point at the correct executable
```
Comment (giuseppeg) [inline tasks/release/osx.js]: nit: "point to" instead of "at"

File: tasks/release/osx.js (hunk 1 above)
Comment (kof) [inline tasks/release/osx.js]: why? what do want to replace with this?

File: tasks/release/osx.js (hunk 2 above)
Comment (kof) [inline tasks/release/osx.js]: funny, its copy paste from the original source and its actually to be removed

File: tasks/release/osx.js (hunk 1 above)
Comment (kof) [inline tasks/release/osx.js]: oh so you would be replacing any char at the beginning which is not a number, that would do it too.

Comment (kof) [inline tasks/release/osx.js]: done, using your regexp now

**Comment by giuseppeg** [comment]
👍

File: docs/build.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+## Common
+
+```bash
+git clone git@github.com:ubergrape/grape-electron.git
+cd grape-electron
+npm install
+npm run release
+```
+The result would be a installable (`dmg` and `app` or `msi` and `exe`) file in `./releases/` with version prefix that match version from `./app/package.json`
```
Comment (giuseppeg) [inline docs/build.md]: I would say

> aN executable installer (`dmg` and `app` or `msi` and `exe`) file in `./releases/` whose name is the `version` field from `./app/package.json`

File: docs/build.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Electron Build Instructions
+
+## Common
+
```
Comment (giuseppeg) [inline docs/build.md]: Maybe add a "warning" to communicate that this process will produce an unsigned app installer which is meant to be a testing version only. The final installer should be signed.

File: docs/build.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+npm install
+npm run release
+```
+The result would be a installable (`dmg` and `app` or `msi` and `exe`) file in `./releases/` with version prefix that match version from `./app/package.json`
+
+## App signing
+
+### MacOS
+
+You need to sign release:
```
Comment (giuseppeg) [inline docs/build.md]: sign A release to:

File: docs/build.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+- to upload it to MacOS AppStore
+- to create a signed release for a distribution without store
+- create and install all certificates from developer.apple.com
+
+  ![](./images/mac-certificates-screen.png)
+
+  http://stackoverflow.com/questions/29039462/which-certificate-should-i-use-to-sign-my-mac-os-x-application
+- Download provision profiles and put them into root:
```
Comment (giuseppeg) [inline docs/build.md]: > the provision

> into the root

you always forget the articles :)

File: docs/build.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  - Electron_Chat_Dev.provisionprofile
+  - Electron_Chat_Prod.provisionprofile
+- Increment [build number](https://github.com/ubergrape/grape-electron/blob/master/app/package.json#L7) for each review upload (you can upload same version multiple times if review failed, but build should be uniq all the time, don't reset this number ever, even for new version)
```
Comment (giuseppeg) [inline docs/build.md]: > the build number

> upload the same version .. if the review failed

> should be uniquE

File: docs/build.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+- Signed store release `npm run release -- --sign {teamId} --mas`
+- Signed distribution outside of store `npm run release -- --sign {teamId}`
+
+For signing you will need group of certificates in `Keychain Access`, described here: https://github.com/electron/electron/blob/master/docs/tutorial/mac-app-store-submission-guide.md#get-certificate
```
Comment (giuseppeg) [inline docs/build.md]: To sign the app you first need to get a certificate from Apple as described here

File: docs/build.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+### Windows
+You need to sign any release on Windows.
+So you need to have certificate on your `win7` and password.
```
Comment (giuseppeg) [inline docs/build.md]: In order to sign you need a certificate and a password.

File: docs/build.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+It is already there on our pre-setup [Parallels Windows7 Image](https://github.com/ubergrape/chatgrape/wiki/Electron-Build-Instructions#windows-1) at `c:\grape_code_signing.p12`
```
Comment (giuseppeg) [inline docs/build.md]: We have a pre configured windows 7 image that you can find at ... The certificate will be in `c:\grape_code_signing.p12`

File: docs/build.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+Or if you're building own environment, you need to get it with password from @sk7
+
+```
+npm run release -- --sign PASSWORD --cert c:\path-to-cert.p12
+```
+
+## macOS
```
Comment (giuseppeg) [inline docs/build.md]: Why is this section empty?

File: docs/build.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+_We're building only 32bit application._
+_To build 64bit app you need to use 64bit Windows._
+
+-----
+
+Otherwise you need to:
+ * install 32bit node (last safe version is `6.3.0`)
+ * install git or download source
+ * install latest [WiX] (http://wixtoolset.org/)
```
Comment (giuseppeg) [inline docs/build.md]: I think that we also need NSIS http://nsis.sourceforge.net/Main_Page

File: docs/build.md (hunk 3 above)
Comment (kof) [inline docs/build.md]: I didn't fix all texts, just changed what needed, most of them a re old and with errors.

File: docs/build.md (hunk 12 above)
Comment (kof) [inline docs/build.md]: no, from what I know only wix, its a replacement for NSIS

Comment (kof) [inline docs/build.md]: need to install all that stuff though to see if those instructions work.

**Comment by giuseppeg** [comment]
looking good

**Comment by kof** [comment]
I think its now finished @giuseppeg

**Comment by kof** [comment]
merged

**Pull request merged**